src/Language/PureScript/Ide/Filter.hs

```diff
@@ -77,6 +79,15 @@ declarationFilter :: (IdeDeclaration -> Text -> Bool) -> Text -> [Module] -> [Mo
 declarationFilter predicate search =
   filterModuleDecls (flip predicate search)
 
+-- | Only keeps Identifiers in the given type declarations
+declarationTypeFilter :: [D.IdeDeclaration] -> Filter
+declarationTypeFilter [] = mkFilter identity
+declarationTypeFilter decls =
+    mkFilter $ filterModuleDecls filterDecls
+    where
+      filterDecls :: IdeDeclaration -> Bool
+      filterDecls decl = D.typeDeclarationForDeclaration decl `elem` decls
```

Comment:
@kRITZCREEK Could you re-start CI builds of travis, please? It seems that there are some `timed out` issues, e.g. https://travis-ci.org/purescript/purescript/jobs/242937235#L667 and here https://travis-ci.org/purescript/purescript/jobs/242937236#L809 

Also I haven't seen the error `Segmentation fault/access violation in generated code` before by running AppVendor: https://ci.appveyor.com/project/hdgarrood/purescript-lfgff/build/2447#L772  Any idea?

Thx :)